mlflow/pyfunc/stdin_server.py

```diff
@@ -0,0 +1,40 @@
+import sys
+import argparse
+import json
+import mlflow
+from mlflow.pyfunc import scoring_server
+import logging
+from pathlib import Path
+
+_logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--model-uri")
+args = parser.parse_args()
+
+_logger.info("Loading model from %s", args.model_uri)
+model = mlflow.pyfunc.load_model(args.model_uri)
+input_schema = model.metadata.get_input_schema()
+_logger.info("Loaded model")
+
+_logger.info("Waiting for request")
+for line in sys.stdin:
+    _logger.info("Received request")
+    request = json.loads(line)
```

Comment:
What if input JSON contains '\n' character ?

Reply:
I think json should escaped it:

```
import json

a = """



"""

print((json.dumps(a),))
```

Output:

```
```

Reply:
I prefer to use binary mode as input / output, and use following protocol,

and for each input, first we send an integer about the input request binary data length, then send the binary data (and then server can decode binary data to JSON )
for output, we can do similar thing.

Reply:
> use the following protocol:

Why do we need this protocol?

Reply:
if we ensure special characters are escaped properly, we can use text input / output

Reply:
Hmmm...our REST scoring server doesn't ensure that special characters are escaped properly, does it? What's the difference here?

Reply:
> our REST scoring server doesn't ensure that special characters are escaped properly,

I think it should handle them properly, otherwise we should hit errors.
we can add test for the case.

Reply:
Got it. Still not seeing what's different about the file approach, compared to the REST scoring approach